sql/xa.cc

```diff
@@ -1196,5 +1196,6 @@ static bool slave_applier_reset_xa_trans(THD *thd)
   thd->has_waiter= false;
   MYSQL_COMMIT_TRANSACTION(thd->m_transaction_psi); // TODO/Fixme: commit?
   thd->m_transaction_psi= NULL;
+  thd->variables.pseudo_thread_id= 0;
```

Comment:
I think a warning in case the pseudo-thread-id is not set is worth of giving, like
```
modified   sql/xa.cc
@@ -1196,6 +1196,10 @@ static bool slave_applier_reset_xa_trans(THD *thd)
   thd->has_waiter= false;
   MYSQL_COMMIT_TRANSACTION(thd->m_transaction_psi); // TODO/Fixme: commit?
   thd->m_transaction_psi= NULL;
+  if (thd->variables.pseudo_slave_mode && thd->variables.pseudo_thread_id == 0)
+    push_warning(thd, Sql_condition::WARN_LEVEL_WARN,
+		 ER_INCORRECT_GLOBAL_LOCAL_VAR,
+		 ER_THD(thd, ER_INCORRECT_GLOBAL_LOCAL_VAR));
   thd->variables.pseudo_thread_id= 0;
   return thd->is_error();
 }

```

Never mind `ER_INCORRECT_GLOBAL_LOCAL_VAR`. I picked only for compile, a new one seems necessary.